src/components/facet-line/FacetLineFilters.tsx

```diff
@@ -38,9 +40,7 @@ const FacetLineFilters: React.FunctionComponent<FacetLineFiltersProps> = ({
     e: React.ChangeEvent<HTMLSelectElement>,
     facet: string
   ) => {
-    const term = facets
-      .find((item) => item.name === facet)
-      ?.values.find((item) => item.key === e.target.value);
+    const term = facetMap[`${facet}:${e.target.value}`][0];
```

Comment:
Two things:
* Why are the values in the map arrays? Isn't it just necessary to have one object?
* Since you made this nice facet map I think it would make sense to make a function that finds the value/object from the given facet/termName. Eg.: `findTermInFacetMap(facetName, termName)`